src/VerifyPIVController.js

```diff
@@ -0,0 +1,125 @@
+/*!
+ * Copyright (c) 2019, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/* eslint complexity:[2, 10], max-params: [2, 11] */
+
+define([
+  'okta',
+  'util/FormController',
+  'util/FormType',
+  'views/shared/FooterWithBackLink'
+],
+function (Okta, FormController, FormType, FooterWithBackLink) {
+
+  var _ = Okta._,
+      $ = Okta.$;
+  var { Util } = Okta.internal.util;
+
+  return FormController.extend({
+    className: 'mfa-verify verify-piv',
+    Model: {
+
+      save: async function () {
+        this.trigger('request');
+        var self = this,
+            data = {
+              fromURI: this.settings.get('relayState')
+            },
+            pivButton = this.settings.get('piv');
+
+        try {
+          await this.getCert(pivButton.certAuthUrl);
+          const res = await this.postCert(pivButton.certAuthUrl, data);
+          Util.redirect(res.redirectUrl);
+        } catch (err) {
```

Comment:
will this `catch` block handle all errors that could happen in `getCert` and `postCert`? As in server is down, maybe  get call times out, etc

Reply:
It should. This test checks that I guess  
https://github.com/okta/okta-signin-widget/pull/932/files#diff-8717f20fe0dfff66217b2c9a59a60190R100

Reply:
Yea verified manually that it catches errors on get as well